File: 프로그래머스/2/49994. 방문 길이/방문 길이.py
def solution(dirs):
    answer = 0
    x, y = 0, 0
    visited = set()
    for dir in dirs:
        if dir == "U" and y+1 <= 5:
            visited.add((x,y, x,y+1))
            visited.add((x,y+1, x,y))
            y += 1
        elif dir == "D" and y-1 >= -5:
            visited.add((x,y, x,y-1))
            visited.add((x,y-1, x,y))
            y -= 1
        elif dir == "R" and x+1 <= 5:
            visited.add((x,y, x+1,y))
            visited.add((x+1,y, x,y))
            x += 1
        elif dir == "L" and x-1 >= -5:
            visited.add((x,y, x-1,y))
            visited.add((x-1,y, x,y))
            x -= 1
        else:
            continue
    answer = len(visited)//2
    return answer

File: 프로그래머스/2/49994. 방문 길이/방문 길이.java
import java.util.*;
class Solution {
    public int solution(String dirs) {
        HashSet<String> answer = new HashSet<>();
        int y = 0, x = 0;
        for (char dir: dirs.toCharArray()) {
            int ny = y, nx = x;
            if (dir == 'U') {
                ny += 1;
            } else if (dir == 'D') {
                ny -= 1;
            } else if (dir == 'L') {
                nx -= 1;
            } else {
                nx += 1;
            }
            if (Math.abs(ny) > 5 || Math.abs(nx) > 5) continue;
            answer.add(y+","+x+","+ny+","+nx);
            answer.add(ny+","+nx+","+y+","+x);
            y = ny;
            x = nx;
        }
        return answer.size() / 2;
    }
}